spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/cassandra/CassandraDataAutoConfiguration.java

```diff
@@ -63,7 +65,7 @@ public class CassandraDataAutoConfiguration {
 
 	private final CqlSession session;
 
-	public CassandraDataAutoConfiguration(CqlSession session) {
+	public CassandraDataAutoConfiguration(@Lazy CqlSession session) {
```

Comment:
I wonder if we should inject and store an `ObjectProvider<CqlSession>` here rather than relying on the proxying that `@Lazy` requires.

Reply:
Relying on the proxy was the less invasive variant.